src/discussions/comments/comment/Comment.jsx

```diff
@@ -39,6 +42,24 @@ function Comment({
   const hasMorePages = useSelector(selectCommentHasMorePages(comment.id));
   const currentPage = useSelector(selectCommentCurrentPage(comment.id));
   const blackoutDateRange = useSelector(selectBlackoutDate);
+  const isUserAdmin = useSelector(selectUserIsStaff);
+  const userHasModerationPrivilages = useSelector(selectUserHasModerationPrivileges);
+  const isUserGroupTA = useSelector(selectUserIsGroupTa);
+  const isCourseAdmin = useSelector(selectIsCourseAdmin);
+  const isCourseStaff = useSelector(selectIsCourseStaff);
+
+  const hasPrivilege = handleAddPostForRoles(isUserAdmin, userHasModerationPrivilages,
+    isUserGroupTA, isCourseAdmin, isCourseStaff);
+
```

Comment:
this utility function can be changed to custom hook and all selectors then can easily be utilized there and hook will be used at all placed needed to calculate user privilege.